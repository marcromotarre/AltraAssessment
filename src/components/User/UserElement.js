import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import HeightSVG from './../../svg/height.svg';
import WeightSVG from './../../svg/weight.svg';
import CandleSVG from './../../svg/candle.svg';
import FriendsSVG from './../../svg/friends.svg';
import ProfessionsSVG from './../../svg/professions.svg';

import UserProperty from './../User/UserProperty.js';
import ColoredHair from './ColoredHair';

const UserContainer = styled.section`
  padding: 10px 30px 10px 10px;
  display: grid;
  grid-template-columns: 58px auto;
  column-gap: 10px;
  border-bottom: 1px solid #e3e3e3;
  background-color: #f8f8f8;
`;

const UserImage = styled.img`
  border-radius: 50%;
  height: 58px;
  width: 58px;
  justify-self: center;
  align-self: center;
`;
const UserInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
`;

const UserName = styled.label`
  width: 100%;
  justify-content: start;
  align-items: center;
  font-family: 'HelveticaNeue-Bold';
  color: #707070;
  font-size: 16px;
`;

const UserProperties = styled.div`
  grid-area: m-properties;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const UserElement = ({
  name,
  thumbnail,
  age,
  weight,
  height,
  hair_color,
  professions,
  friends,
}) => {
  return (
    <UserContainer>
      <UserImage data-testid="user-image" src={thumbnail} />
      <UserInfo>
        <UserName data-testid="user-name">{name}</UserName>
        <UserProperties data-testid="user-properties">
          <UserProperty>
            <CandleSVG width={14.63} /> <label>{age}</label>
          </UserProperty>
          <UserProperty data-testid="user-height">
            <HeightSVG height={17.7} />
            <label>
              {isNaN(height) ? height : Math.floor(height).toFixed(0)}
            </label>
          </UserProperty>
          <UserProperty data-testid="user-width">
            <WeightSVG width={17.7} />
            <label>
              {isNaN(weight) ? weight : Math.floor(weight).toFixed(0)}
            </label>
          </UserProperty>
          <UserProperty>
            <ColoredHair color={hair_color} width={15.26} />
          </UserProperty>
          <UserProperty>
            <ProfessionsSVG width={14.34} />
            <label>{professions.length}</label>
          </UserProperty>
          <UserProperty>
            <FriendsSVG width={15.26} /> <label>{friends.length}</label>
          </UserProperty>
        </UserProperties>
      </UserInfo>
    </UserContainer>
  );
};

UserElement.propTypes = {
  name: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  weight: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  hair_color: PropTypes.string.isRequired,
  professions: PropTypes.arrayOf(PropTypes.string).isRequired,
  friends: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default UserElement;
