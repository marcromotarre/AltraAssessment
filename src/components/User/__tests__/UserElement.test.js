import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import UserElement from '../UserElement';

/** this component should:
 *  - render all user data
 */

describe('User Element Component tests', () => {
  const props = {
    name: 'user name',
    thumbnail: 'imagepath',
    age: 20,
    weight: 70.76543,
    height: 150.83749,
    hair_color: 'Green',
    professions: ['profession1', 'profession2'],
    friends: ['friend1', 'friend2', 'friend3'],
  };

  it('render UserElement Snapshot', () => {
    const tree = renderer.create(<UserElement {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render user name', () => {
    const { getByTestId } = render(<UserElement {...props} />);
    const userName = getByTestId('user-name');
    expect(userName.innerHTML).toBe(props.name);
  });

  it('should render all properties name', () => {
    const { getByTestId } = render(<UserElement {...props} />);
    const userProperties = getByTestId('user-properties');
    expect(userProperties.children.length).toBe(6);

    const age = userProperties.children[0];
    expect(age.children.length).toBe(2);
    expect(age.querySelector('svg').getAttribute('width')).toBe('14.63');
    expect(age.querySelector('label').innerHTML).toBe(`${props.age}`);

    const height = userProperties.children[1];
    expect(height.children.length).toBe(2);
    expect(height.querySelector('svg').getAttribute('height')).toBe('17.7');
    expect(height.querySelector('label').innerHTML).toBe('150');

    const weight = userProperties.children[2];
    expect(weight.children.length).toBe(2);
    expect(weight.querySelector('svg').getAttribute('width')).toBe('17.7');
    expect(weight.querySelector('label').innerHTML).toBe('70');

    const hairColor = userProperties.children[3];
    expect(hairColor.children.length).toBe(1);
    expect(hairColor.querySelector('svg').getAttribute('width')).toBe('15.26');
    expect(hairColor.querySelector('svg').getAttribute('color')).toBe(
      props.hair_color
    );

    const professions = userProperties.children[4];
    expect(professions.children.length).toBe(2);
    expect(professions.querySelector('svg').getAttribute('width')).toBe(
      '14.34'
    );
    expect(professions.querySelector('label').innerHTML).toBe(
      `${props.professions.length}`
    );

    const friends = userProperties.children[5];
    expect(friends.children.length).toBe(2);
    expect(friends.querySelector('svg').getAttribute('width')).toBe('15.26');
    expect(friends.querySelector('label').innerHTML).toBe(
      `${props.friends.length}`
    );
  });
});
