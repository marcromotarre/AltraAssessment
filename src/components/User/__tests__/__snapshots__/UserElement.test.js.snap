// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Element Component tests render UserElement Snapshot 1`] = `
<section
  className="sc-AxiKw gZriPm"
>
  <img
    className="sc-AxhCb eTSSEI"
    data-testid="user-image"
    src="imagepath"
  />
  <div
    className="sc-AxhUy QOkiE"
  >
    <label
      className="sc-AxgMl ijardh"
      data-testid="user-name"
    >
      user name
    </label>
    <div
      className="sc-AxheI jYVyby"
      data-testid="user-properties"
    >
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          width={14.63}
        />
         
        <label>
          20
        </label>
      </div>
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          height={17.7}
        />
        <label>
          150
        </label>
      </div>
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          width={17.7}
        />
        <label>
          70
        </label>
      </div>
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          className="sc-AxirZ itfaLk"
          color="Green"
          width={15.26}
        />
      </div>
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          width={14.34}
        />
        <label>
          2
        </label>
      </div>
      <div
        className="sc-AxjAm cDIZLj"
        data-testid="user-property"
      >
        <svg
          width={15.26}
        />
         
        <label>
          3
        </label>
      </div>
    </div>
  </div>
</section>
`;
